library/choose_disk.py

```diff
@@ -361,6 +363,17 @@ def disk_label(partition):
     if "ceph journal" in stdout:
         return "journal"
 
+    # Then, let's search for metadata coming from ceph-volume
+    output_cmd = subprocess.Popen("ceph-volume lvm list --format=json {}".format(partition),
+                                  shell=True, stdout=subprocess.PIPE)
+    raw_json, _ = output_cmd.communicate()
+    json = json.loads(raw_json)
+    for item in json:
+        current_item = json[item]
+        if "type" in current_item[0]:
```

Comment:
this is only looking at the `0` index, which means it is assuming there will always be items in that list. Not entirely sure when/if we will see an empty JSON for a device, but it would be just safer to loop over the list and retrieve the values that you need

Reply:
I agree that if the output is empty, it have to be handled : fixed.
As I'm parsing the disk directly, it would be erratic have multiple entries. I've added a check for this case and parse the sole entry I should get.

Reply:
And as its a list, I cannot access it without an integer index.